### ISSUE TITLE
Clarify remote kernel connection process (how to find connection file and enter credentials)

### ISSUE THREAD

**Issue by ccordoba12** [spyder-ide/spyder-docs#78]
@jsh9 commented on [Sat Sep 08 2018](https://github.com/spyder-ide/spyder/issues/7861)

## Problem Description
I am following the steps in the documentation (https://docs.spyder-ide.org/ipythonconsole.html#connect-to-an-external-kernel) to try to open up an ipython kernel on my remote computer and then access it from my local computer.

I used `ipython kernel` on my remote computer, and saw the following message on the terminal:
```
NOTE: When using the `ipython kernel` entry point, Ctrl-C will not work.

To exit, you will have to explicitly quit this process, by either sending
"quit" from a client, or using Ctrl-\ in UNIX-like environments.

To read more about this, see https://github.com/ipython/ipython/issues/2049


To connect another client to this kernel, use:
    --existing kernel-17928.json
```
This step in the documentation ("Make sure to copy the connection file (runtime-dir/kernel-pid.json) to the machine you’re running Spyder on (if remote) or note its location (if local).") requires me to find the location of `kernel-17928.json`, but I don't know where this json file is on my remote computer.

## Versions

* Spyder version: 3.3.1 (remote) and 3.3.0 (local)
* Python version: 3.6
* Qt version: 5.9.6 (remote) and 5.6.2 (local)
* PyQt version: 5.9.2 (remote) and 5.6.0 (local)
* Operating System name/version: macOS (remote), Windows (local)

---

@ccordoba12 commented on [Sat Sep 08 2018](https://github.com/spyder-ide/spyder/issues/7861#issuecomment-419690891)

Moving to the right repo.

**Comment by ccordoba12** [comment]
@jsh9, this is a shortcoming in our docs. Thanks for pointing it out.

@CAM-Gerlach, we need to address this. You can get `runtime-dir` with this code:

```ipython
from jupyter_core import paths
paths.jupyter_runtime_dir()
```

**Comment by CAM-Gerlach** [comment]
@ccordoba12 Thanks! I can also address the related remote kernel connection confusion that was just brought up in our public Gitter chat:

> Hi, I am trying to connect Spyder on my PC (running within Anaconda) to a remote ipython kernel on an AWS EC2 AMI and I'm struggling. I followed the instructions here: http://docs.spyder-ide.org/ipythonconsole.html#connect-to-an-external-kernel and the kernel is running fine on the EC2 AMI and I have the JSON file on my PC but I'm not sure what to enter in the 'Connect to an existing kernel' box. I add the path to the JSON file in the 'Connection info' box. I check the 'This is a remote kernel' box. I enter my EC2 host: 'ec2-user@blah-blah:8888' and then get stuck on the SSH key and password boxes. When I connect to the EC2 AMI from MobaXterm I use an imported Putty key (.ppk) file (and don't use a password) whereas the Spyder dialogue box is asking for a .pem file. Do I need the original AWS .pem key?

**Comment by jsh9** [comment]
Thank you @ccordoba12 !  I found the file on the path, downloaded it to my local machine, and now I can connect to the remote kernel.

It might also be worth noting (either in Spyder's UI, or in the documentation) that the "ssh key" field is optional.  I spent quite some time googling what ssh key file I should provide to no avail.  Other users may run into the same confusion.

**Comment by CAM-Gerlach** [comment]
@jsh9 Thanks; glad you got it working.

> It might also be worth noting (either in Spyder's UI, or in the documentation) that the "ssh key" field is optional. I spent quite some time googling what ssh key file I should provide to no avail. Other users may run into the same confusion.

In fact, in my comment immediately before yours, I described exactly that. Ideally it should be clearly indicated in the UI (such as by including the text "Or" between the two fields and grouping them in some way, or at least including explanatory text in the dialog itself; @ccordoba12 I can open an issue if you want), but it would be  good to have in the docs too, which is why I included it here and changed the title accordingly. Thanks!
